jwst/refpix/reference_pixels.py

```diff
@@ -695,6 +755,61 @@ def do_corrections(self):
         log.setLevel(logging.INFO)
         return
 
+    def do_subarray_corrections(self):
+        """Do corrections for subarray.  Reference pixel value calculated
+        separately for odd and even columns if odd_even_columns is True,
+        otherwise a single number calculated from all reference pixels"""
+        #
+        #  First transform to detector coordinates
+        #
+        self.DMS_to_detector()
+        refdq = dqflags.pixel['REFERENCE_PIXEL']
+        (nints, ngroups, nrows, ncols) = self.data.shape
+        for integration in range(nints):
+            for group in range(ngroups):
+                #
+                # Get the reference values from the top and bottom reference
+                # pixels
+                #
+                thisgroup = self.data[integration, group].copy()
+                refpixindices = np.where(np.bitwise_and(self.pixeldq, refdq) == refdq)
```

Comment:
Where is it in the upstream processing that we're assuming the ``REFERENCE_PIXEL`` DQ flags got added to the ``pixeldq`` array? Are they supposed to come from the MASK reference file that gets loaded by the ``dq_init`` step? I don't recall if all MASK ref files have yet been updated to include those flags. Are the checks below sufficient to keep things from blowing up if those flags haven't been set?

Reply:
Yes, I'm assuming they get added to the pixeldq array from the dq_init step.  If the flags haven't been set, the step should be skipped.